### ISSUE TITLE
Video output broken if both command line player and libmpv are enabled [was: No video output due to OpenGL error INVALID_FRAMEBUFFER_OPERATION]

### ISSUE THREAD

**Issue by areinisc** [mpv-player/mpv#1579]
Not sure if this is a problem with mpv or with my setup. Playing almost any file-type results in output like:
```
Playing: sample-10s.mkv
[stream] Video (+) --vid=1 (h264)
[stream] Audio (+) --aid=1 --alang=eng (*) (mp3)
AO: [coreaudio] 48000Hz stereo 2ch s16
VO: [opengl] 704x400 yuv420p
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
AV: 00:00:00 / 00:00:15 (0%) A-V:  0.000
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
AV: 00:00:00 / 00:00:15 (0%) A-V:  0.000
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
AV: 00:00:00 / 00:00:15 (0%) A-V:  0.000
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
AV: 00:00:00 / 00:00:15 (1%) A-V:  0.000
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
AV: 00:00:15 / 00:00:15 (99%) A-V:  0.000


Exiting... (End of file)
```
There should be video output and I'm uncertain why there isn't any. I haven't found any useful information after searching around with the error output.

Here is the output from `brew info mpv`:
```
mpv: stable 0.7.3, HEAD
https://github.com/mpv-player/mpv
/usr/local/Cellar/mpv/0.7.3 (360 files, 35M) *
  Built from source with: --with-libaacs, --with-libbluray, --with-libdvdnav, --with-libdvdread, --with-libmpv, --with-x11
From: https://github.com/mpv-player/homebrew-mpv/blob/master/Formula/mpv.rb

[Dependencies and Options listings redacted]

==> Caveats
mpv.app installed to:
  /usr/local/Cellar/mpv/0.7.3

To link the application to a normal Mac OS X location:
    brew linkapps
or:
    ln -s /usr/local/Cellar/mpv/0.7.3/mpv.app /Applications

zsh completion has been installed to:
  /usr/local/share/zsh/site-functions

.app bundles were installed.
Run `brew linkapps mpv` to symlink these to /Applications.
```

I've put an mkv, mp4, and 3gp that all reproduce the error I see at [this ge.tt](http://ge.tt/4n6wBGA2) for you to use as samples.

Thanks for your help in advance!

**Comment by pigoz** [comment]
What if you remove `--with-x11`?

**Comment by wm4** [comment]
There's nothing special with these files; it's just a (yet) unknown problem with opengl. Log with ``-v`` would be useful.

**Comment by Nyx0uf** [comment]
I recall having this problem, disabling building of libmpv fixed it (iMac 2012, OS X 10.10)

It was from git master not homebrew though

**Comment by areinisc** [comment]
@wm4 Sorry, I'm not sure what you mean by "Log with `-v` would be useful." In any case, I just ran `mpv log -v > mpvlog-v.txt` and `mpv -v > mpv-v.txt` and made a [gist with the output here][href1]. Please let me know what to do if that's not what you wanted.

[href1]: https://gist.github.com/areinisc/06f096edc80d78a16070 "gist of mpv troubleshooting logs"

**Comment by wm4** [comment]
I meant: run ``mpv somefile.mkv -v > log.txt`` (just for some seconds)

**Comment by areinisc** [comment]
@pigoz Ok, here's what I did:
```
brew uninstall mpv
brew update
brew install mpv --with-libaacs --with-libbluray --with-libdvdnav --with-libdvdread --with-libmpv
brew linkapps mpv
```
and then running `mpv sample-10s.mkv` gives the following (looks the same as before) output:
```
Playing: sample-10s.mkv
[stream] Video (+) --vid=1 (h264)
[stream] Audio (+) --aid=1 --alang=eng (*) (mp3)
AO: [coreaudio] 48000Hz stereo 2ch s16
VO: [opengl] 704x400 yuv420p
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
AV: 00:00:00 / 00:00:15 (0%) A-V:  0.000
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
AV: 00:00:00 / 00:00:15 (0%) A-V:  0.000
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
AV: 00:00:00 / 00:00:15 (0%) A-V:  0.000
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
AV: 00:00:00 / 00:00:15 (1%) A-V:  0.000
[vo/opengl] after rendering: OpenGL error INVALID_FRAMEBUFFER_OPERATION.
AV: 00:00:15 / 00:00:15 (99%) A-V:  0.000


Exiting... (End of file)
```
Still no video output.

Is this what you had in mind? Is there something else I should do to test what you suggested?

**Comment by areinisc** [comment]
@wm4 whoops! I updated [the gist][href1] to have the contents of `log.txt`after running `mpv sample-10s.mkv -v > log.txt`. (Also: before doing this I uninstalled mpv and reinstalled with my original flags (including --with-x11).)

[href1]: https://gist.github.com/areinisc/06f096edc80d78a16070 "mpv troubleshooting gist"

**Comment by wm4** [comment]
Thanks, this is what I wanted. I still don't know why it fails though. I'll take a closer look tomorrow.

>I recall having this problem, disabling building of libmpv fixed it (iMac 2012, OS X 10.10)

@Nyx0uf: this shouldn't happen. Maybe there's something very wrong here, like the cocoa part being broken. This issue might be exactly the same problem.

**Comment by areinisc** [comment]
@Nyx0uf Nice catch! 

I ran:
```
brew uninstall mpv
brew update
brew install mpv --with-libaacs --with-libbluray --with-libdvdnav --with-libdvdread --with-x11
brew linkapps mpv
mpv sample-10s.mkv
```
and got video (!) along with:
```
Playing: sample-10s.mkv
[stream] Video (+) --vid=1 (h264)
[stream] Audio (+) --aid=1 --alang=eng (*) (mp3)
AO: [coreaudio] 48000Hz stereo 2ch s16
VO: [opengl] 704x400 yuv420p
AV: 00:00:15 / 00:00:15 (99%) A-V:  0.000


Exiting... (End of file)
```

I don't know what libmpv is or does, so I don't *really* know where to go from here. I assume that this fixes the issue for me and I should just leave it this way, but if that's not the best thing to do I'm all ears. Also, if I would be helping in some way, please just let me know what to do/run/try and I'll give it a whirl. 
Thanks for all the help!

**Comment by wm4** [comment]
This should really be fixed at some point. (The easy way would be to make cplayer and libmpv exclusive choices.)

**Comment by wm4** [comment]
Fixed in 19a5b20752ecc7465cf17781f908e12bf4ca136d.
